### PR TITLE
feature: add without base option for sealer build

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

Applications Image contains applications with all dependencies except the base image, so ApplicationsImage can install to an already existing Kubernetes cluster.

add "--no-base" option for sealer build, if it is true, will only keep the application context and its docker image.

in order to keep the same with the "--no-cache" option. name it as "--no-base".

run this cmd will build app image:`sealer build -f Kubefile -t myapp:v1 -m lite . --no-base`
if want keep all images include base image, run `sealer build -f Kubefile -t myimage:v1 -m lite .`

### Special notes for reviews
link to issue #695

### CODE REVIEW

File: build/buildkit/buildimage/build_image.go [hunk 1]
```diff
@@ -119,7 +120,37 @@ func (b BuildImage) genNewLayer(layerType, layerValue, filepath string) (v1.Laye
 	return imageLayer, nil
 }
 
-func (b BuildImage) SaveBuildImage(name string) error {
+func (b BuildImage) checkImageMataData(opts SaveOpts) error {
```
Comment (fanux) [inline build/buildkit/buildimage/build_image.go:123]: Metadata

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -33,9 +33,10 @@ type Interface interface {
 }
 
 type Metadata struct {
-	Version string `json:"version"`
-	Arch    string `json:"arch"`
-	Variant string `json:"variant"`
+	Version     string `json:"version"`
+	Arch        string `json:"arch"`
+	Variant     string `json:"variant"`
+	KubeVersion string `json:"kubeVersion"`
```
Comment (fanux) [inline pkg/runtime/runtime.go:39]: What difference to Version?

File: sealer/cmd/build.go [hunk 3]
```diff
@@ -74,7 +81,9 @@ func init() {
 	buildCmd.Flags().StringVarP(&buildConfig.BuildType, "mode", "m", "lite", "cluster image build type, default is lite")
 	buildCmd.Flags().StringVarP(&buildConfig.KubefileName, "kubefile", "f", "Kubefile", "kubefile filepath")
 	buildCmd.Flags().StringVarP(&buildConfig.ImageName, "imageName", "t", "", "cluster image name")
+	buildCmd.Flags().StringVarP(&buildConfig.KubeVersion, "kubeversion", "k", "", "KubeVersion is a SemVer constraint specifying the version of Kubernetes required.")
```
Comment (fanux) [inline sealer/cmd/build.go:84]: Why needs this argument?

File: pkg/runtime/runtime.go (hunk 2 above)
Comment (kakaZhou719) [reply pkg/runtime/runtime.go:39]: use KubeVersion to save version constraint.

File: sealer/cmd/build.go (hunk 3 above)
Comment (kakaZhou719) [reply sealer/cmd/build.go:84]: i think user could specify the version constraint

**Comment by fanux** [comment]
I think `--base false` is better, default is true.

**Review by fanux — APPROVED** [review]

**Pull request merged**